src/main/java/htsjdk/variant/variantcontext/GenotypeLikelihoods.java

```diff
@@ -471,25 +464,24 @@ else if (ploidy == 1)
      *
      * Note this method caches the value for most common num Allele / ploidy combinations for efficiency
      *
-     * Recursive implementation:
-     *   S(N,P) = sum_{k=0}^P S(N-1,P-k)
-     *  because if we have N integers, we can condition 1 integer to be = k, and then N-1 integers have to sum to P-K
-     * With initial conditions
-     *   S(N,1) = N  (only way to have N integers add up to 1 is all-zeros except one element with a one. There are N of these vectors)
-     *   S(1,P) = 1 (only way to have 1 integer add to P is with that integer P itself).
+     * For non-cached values, the result is calculated via a call to calcNumLikelihoods,
+     * which uses the Apache Commons CombinatoricsUtils class
+     * using the formula (numAlleles + ploidy - 1) choose ploidy
      *
      *   @param  numAlleles      Number of alleles (including ref)
      *   @param  ploidy          Ploidy, or number of chromosomes in set
      *   @return    Number of likelihood elements we need to hold.
      */
     public static int numLikelihoods(final int numAlleles, final int ploidy) {
```

Comment:
now that the cache is dynamic this needs to be synchronized or we risk weird intermediate states.

Reply:
Good catch; I'll add that